### PR TITLE
more misc changes sorry

### PR DESCRIPTION
**Pull request by PeterMitrano**
1. improve some exceptions & error handling
2. provides access to the new C++ API I added
3. fixings an annoying issue with getting joint names

### CODE REVIEW

File: arm_robots/src/arm_robots/robot.py [hunk 1]
```diff
@@ -342,10 +407,21 @@ def follow_jacobian_to_position(self,
...
-        return self.get_move_group_commander(group_name).get_active_joints()
+    def get_joint_names(self, group_name: Optional[str] = None):
+        # NOTE: Getting a list of joint names should not require anything besides a lookup on the ros parameter server.
+        #   However, MoveIt does not have this implemented that way, it requires connecting to the move group node.
+        #   This causes problems sometimes, so to simplify things we do the lookup manually if no group name is given.
+        if group_name is None:
+            robot = urdf.from_parameter_server()
+            active_joint_names = [j.name for j in robot.joints if j.type != 'fixed']
+            return active_joint_names
+        else:
```
Comment (bsaund) [inline arm_robots/src/arm_robots/robot.py:418]: `else` not needed

File: arm_robots/src/arm_robots/robot.py [hunk 2]
```diff
@@ -342,10 +407,21 @@ def follow_jacobian_to_position(self,
...
+            if hasattr(self.robot_commander, 'get_active_joint_names'):
+                return self.robot_commander.get_active_joint_names(group_name)
+            else:
```
Comment (bsaund) [inline arm_robots/src/arm_robots/robot.py:421]: `else` not needed

File: arm_robots/src/arm_robots/robot.py [hunk 3]
```diff
@@ -342,10 +407,21 @@ def follow_jacobian_to_position(self,
...
+                return self.get_move_group_commander('whole_body').get_active_joints()
```
Comment (bsaund) [inline arm_robots/src/arm_robots/robot.py:422]: As currently written, in this last case `group_name` is silently ignored. Pretty confusing. I'd rather this throw an error than return the wrong set of joints

File: arm_robots/src/arm_robots/robot.py [hunk 4]
```diff
@@ -295,6 +306,65 @@ def store_tool_orientations(self, preferred_tool_orientations: Optional[Dict]):
     def store_current_tool_orientations(self, tool_names: Optional[List]):
         self.stored_tool_orientations = {n: self.get_orientation(n) for n in tool_names}
 
+    def get_preferred_tool_orientations(self, tool_names: List[str],
+                                        preferred_tool_orientations: Optional[List] = STORED_ORIENTATION):
```
[the remark this answers is not in the document]
Comment (PeterMitrano) [reply arm_robots/src/arm_robots/robot.py:310]: hm would `merge_preferred_tool_orientations` be a better name? or is there just a better way we could implement this?

File: arm_robots/src/arm_robots/robot.py (hunk 3 above)
Comment (PeterMitrano) [reply arm_robots/src/arm_robots/robot.py:422]: oh I think that's just a mistake, it should  use group_name

File: arm_robots/src/arm_robots/robot.py [hunk 5]
```diff
@@ -295,6 +306,65 @@ def store_tool_orientations(self, preferred_tool_orientations: Optional[Dict]):
...
+            tool_names=tool_names,
+            preferred_tool_orientations=preferred_tool_orientations,
+            start_state=robot_state,
+            scene=scene_msg,
+            grippers=points,
+            max_velocity_scaling_factor=vel_scaling,
+            max_acceleration_scaling_factor=0.1,
+        )
+
+        planning_success = reached
```
[the remark this answers is not in the document]
Comment (PeterMitrano) [reply arm_robots/src/arm_robots/robot.py:359]: I'm gonna leave this as is if that's alright, I anticipate using these variables separately in the future

**Pull request merged**